Surge FX build error on Linux when compiling with AVX2 support

**Bug Description:**
Compile error:

```
src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp: In member function ‘void chowdsp::SpringReverbProc::setParams(const chowdsp::SpringReverbProc::Params&, int)’:
src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp:85:75: error: cannot convert ‘__m128’ to ‘juce::dsp::SIMDRegister<float>’
apf.setParams(msToSamples(0.35f + 3.0f * params.size), _mm_load_ps(apfGVec));
                                                       ~~~~~~~~~~~^~~~~~~~~
                                                                  |
                                                                  __m128
```

**Surge XT Version**
1.0.1 release, also tried HEAD 5e81fc20bb22a891c039aaf2951d6454d86694b4

**Reproduction Steps:**
1. do "export CXXFLAGS=-mavx2" and compile as usual

**Additional Information:**

Tried with gcc-11 and clang-13. Clang gives a more verbose error description.

Any -march=XXX which includes AVX2, such as "-march=haswell", or "-march=native" on a modern cpu, will trigger the error.

Compiling with just "-mavx" works, also with any -march if "-mno-avx2" is added.

Looks to me like juce::dsp::SIMDRegister<float> becomes __m256 when avx2 is supported, so the apfGVec should be an array of 8 floats, and _mm256_load_ps should be used instead?

But then regular non-avx builds would break so... we'd have to wrap everything into an `#ifdef __AVX__`. Or just force the SSE/non-avx code path for this particular function.

--- a/src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp
+++ b/src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp
@@ -83,7 +83,7 @@ void SpringReverbProc::setParams(const Params &params, int numSamples)
     auto apfG = 0.5f - 0.4f * params.spin;
     float apfGVec alignas(16)[4] = {apfG, -apfG, apfG, -apfG};
     VecType apfGVecAsVecType = 0;
-    for (int i=0; i<4; ++i)
+    for (int i = 0; i < 4; ++i)
         apfGVecAsVecType[i] = apfGVec[i];
     for (auto &apf : vecAPFs)
         apf.setParams(msToSamples(0.35f + 3.0f * params.size), apfGVecAsVecType);

--- a/src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp
+++ b/src/common/dsp/effects/chowdsp/spring_reverb/SpringReverbProc.cpp
@@ -82,8 +82,11 @@ void SpringReverbProc::setParams(const Params &params, int numSamples)
 
     auto apfG = 0.5f - 0.4f * params.spin;
     float apfGVec alignas(16)[4] = {apfG, -apfG, apfG, -apfG};
+    VecType apfGVecAsVecType = 0;
+    for (int i=0; i<4; ++i)
+        apfGVecAsVecType[i] = apfGVec[i];
     for (auto &apf : vecAPFs)
-        apf.setParams(msToSamples(0.35f + 3.0f * params.size), _mm_load_ps(apfGVec));
+        apf.setParams(msToSamples(0.35f + 3.0f * params.size), apfGVecAsVecType);
 
     constexpr float dampFreqLow = 4000.0f;
     constexpr float dampFreqHigh = 18000.0f;

